### PR TITLE
Studio2Manager Publish

### PR DESCRIPTION
**Pull request by kanak8278**
* Implemented Publish Modal
* API Call to v1/install endpoint from frontend
* Fetching all the keys required to publish bot from DSL

### CODE REVIEW

File: studio/src/pages/editor-page.tsx [hunk 1]
```diff
@@ -414,17 +415,15 @@ export const EditorPage: React.FunctionComponent = () => {
...
-                                                    key: 'download',
-                                                    name: 'Download Transcripts',
-                                                    iconProps: { iconName: 'Download' },
-                                                    onClick: () => {},
-                                                }, {
-                                                    key: 'callbackForm',
-                                                    name: 'Callback form',
-                                                    iconProps: { iconName: 'FormLibrary' },
-                                                    onClick: () => {  },
+                                                    key: 'clearData',
```
Comment (sameersegal) [inline studio/src/pages/editor-page.tsx:421]: Don't remove Download Transcripts and Callback Forms

File: studio/src/components/PublishModal/index.tsx [hunk 2]
```diff
@@ -1,11 +1,54 @@
...
 interface props {
-    isOpen: boolean
-    hideModal: Function
+    isOpen: boolean,
+    hideModal: Function,
+    representations: any,
+    dslName: string
 }
+
+const uploadProgram = (url: string, secret: string, name: string, representations: any) => {
```
Comment (sameersegal) [inline studio/src/components/PublishModal/index.tsx:13]: This function's arguments list:

'name': 
      'dsl': ,
      'code': ,
      'requirements': '',
      'required_credentials': ,
      'index_urls':
      'version':

Comment (sameersegal) [reply studio/src/components/PublishModal/index.tsx:13]: Preprocessing needs to happen before

Comment (sameersegal) [reply studio/src/components/PublishModal/index.tsx:13]: While this is ideal, let the current pattern exists because I want to move this to the backend.

**Review by sameersegal — CHANGES_REQUESTED** [review]

File: studio/src/pages/editor-page.tsx (hunk 1 above)
Comment (kanak8278) [reply studio/src/pages/editor-page.tsx:421]: image

They are back.

**Review by sameersegal — APPROVED** [review]

**Pull request merged**